prov/rxm/src/rxm_cq.c

```diff
@@ -609,11 +609,11 @@ static ssize_t rxm_handle_rndv(struct rxm_rx_buf *rx_buf)
 		}
 	} else {
 		struct rxm_mr *mr;
-
 		for (i = 0; i < rx_buf->recv_entry->rxm_iov.count; i++) {
 			mr = rx_buf->recv_entry->rxm_iov.desc[i];
+			rx_buf->mr[i] = mr->msg_mr;
 			rx_buf->recv_entry->rxm_iov.desc[i] =
-				fi_mr_desc(mr->msg_mr);
+				fi_mr_desc(rx_buf->mr[i]);
```

Comment:
Can you add more details to the commit message regarding what is broken?

Reply:
taking another look at this, I think the patch is actually not needed and PR can be closed.

the `rx_buf->mr` is later closed, the issue I was attempting to fix was correcting the pointers stored here so that we're closing the proper memory regions.  That cleanup, however, only happens with local rdm memory regions, which would not come into this branch of code.